joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/parser/RubyNodeCreator.scala

```diff
@@ -774,6 +774,11 @@ class RubyNodeCreator extends RubyParserBaseVisitor[RubyNode] {
           x
         }
 
+        val (allowedTypeDeclChildren, nonAllowedTypeDeclChildren) = rest.partition {
```

Comment:
```suggestion
        val (allowedTypeDeclChildren, nonAllowedTypeDeclChildren) = rest.partition(_.isInstanceOf[AllowedTypeDeclarationChild])
```

Reply:
Given where this is being filtered, it may be filtering out `AliasStatements` before they are lowered. Perhaps this partitioning should happen right before the `TypeDecl` node is repacked